src/project/view/recentprojectsmodel.cpp

```diff
@@ -34,9 +36,31 @@ using namespace mu::notation;
 namespace {
 const QString SCORE_NAME_KEY("name");
 const QString SCORE_PATH_KEY("path");
-const QString SCORE_THUMBNAIL_KEY("thumbnail");
+const QString SCORE_THUMBNAIL_PIXMAP_KEY("thumbnailPixmap");
+const QString SCORE_THUMBNAIL_URL_KEY("thumbnailUrl");
 const QString SCORE_TIME_SINCE_MODIFIED_KEY("timeSinceModified");
 const QString SCORE_ADD_NEW_KEY("isAddNew");
+
+const QString THUMBNAIL_PLACEHOLDER_MIDI_URL("qrc:/resources/Placeholder_MIDI.png");
+const QString THUMBNAIL_PLACEHOLDER_MXML_URL("qrc:/resources/Placeholder_MXML.png");
+const QString THUMBNAIL_PLACEHOLDER_OTHER_URL("qrc:/resources/Placeholder_Other.png");
+
+QString thumbnailPlaceholderUrlBySuffix(const std::string suffix)
```

Comment:
This code doesn't look like it belongs to this file. Ideally, RecentProjectsProvider should provide the meta info object from which we can directly get the thumbnail without any additional stuff in this model

Reply:
Also, this function should be static. And you missed &

Reply:
Moved it to QML. (The only extra thing that the model does now, is providing the suffix, so that QML does not need to figure out the suffix from the path.)